pkg/provisioner/crdprovisioner.go

```diff
@@ -600,29 +647,36 @@ func (c *CrdProvisioner) UnpublishVolume(
 	nodeID string,
 	secrets map[string]string,
 	mode consts.UnpublishMode) error {
-
+	var err error
 	azVAClient := c.azDiskClient.DiskV1beta1().AzVolumeAttachments(c.namespace)
 	azVALister := c.conditionWatcher.informerFactory.Disk().V1beta1().AzVolumeAttachments().Lister().AzVolumeAttachments(c.namespace)
 
-	volumeName, err := azureutils.GetDiskName(volumeID)
+	var volumeName string
+	volumeName, err = azureutils.GetDiskName(volumeID)
 	if err != nil {
 		return err
 	}
 
 	attachmentName := azureutils.GetAzVolumeAttachmentName(volumeName, nodeID)
 	volumeName = strings.ToLower(volumeName)
 
-	azVolumeAttachmentInstance, err := azVALister.Get(attachmentName)
+	var azVolumeAttachmentInstance *diskv1beta1.AzVolumeAttachment
+	azVolumeAttachmentInstance, err = azVALister.Get(attachmentName)
+	ctx, workflow := azureutils.StartWorkflow(ctx, "CrdProvisioner.UnpublishVolume", azureutils.GetObjectDetails(azVolumeAttachmentInstance)...)
+	defer func() { workflow.Finish(err) }()
+
 	if err != nil {
 		if apiErrors.IsNotFound(err) {
-			klog.Infof("AzVolumeAttachment(%s) deleted: volume (%s) has been successfully detached from node (%s)", attachmentName, volumeName, nodeID)
+			workflow.Logger().WithValues(consts.VolumeNameLabel, volumeName, consts.NodeNameLabel, nodeID).Info("Volume detach successful")
```

Comment:
Weren't volume and node name already added by `azureutils.GetObjectDetails()`?

Reply:
Same as for when `AzVolume` is not found, when `AzVolumeAttachment` is not found, logger needs to be manually added values.